x-pack/legacy/plugins/ml/server/models/results_service/results_service.js

```diff
@@ -402,11 +402,114 @@ export function resultsServiceProvider(callWithRequest) {
     return definition;
   }
 
+  function getFieldAgg(fieldType, query) {
+    const AGG_SIZE = 100;
+
+    const fieldNameKey = `${fieldType}_name`;
+    const fieldValueKey = `${fieldType}_value`;
+
+    return {
+      [fieldNameKey]: {
+        terms: {
+          field: fieldNameKey,
+        },
+      },
+      [fieldValueKey]: {
+        filter: {
+          wildcard: {
+            [fieldValueKey]: {
+              value: query ? `*${query}*` : '*',
+            },
+          },
+        },
+        aggs: {
+          values: {
+            terms: {
+              size: AGG_SIZE,
+              field: fieldValueKey,
+            },
+          },
+        },
+      },
+    };
+  }
+
+  function getFieldObject(field, aggs) {
+    const fieldNameKey = `${field}_name`;
+    const fieldValueKey = `${field}_value`;
+
+    return aggs[fieldNameKey].buckets.length > 0
+      ? {
+          [field]: {
+            name: aggs[fieldNameKey].buckets[0].key,
+            values: aggs[fieldValueKey].values.buckets.map(({ key }) => key),
+          },
+        }
+      : {};
+  }
+
+  /**
+   * Gets the record of partition fields with possible values that fit the provided queries.
+   * @param {Object} searchTerm - object of queries for partition fields, e.g. { partition_field: 'query' }
+   * @param {Object} criteriaFields - key - value pairs of the term field, e.g. { job_id: 'ml-job', detector_index: 0 }
+   * @param {number} earliestMs
+   * @param {number} latestMs
+   * @returns {Promise<*>}
+   */
+  async function getPartitionFieldsValues(searchTerm = {}, criteriaFields, earliestMs, latestMs) {
+    const fields = ['partition_field', 'over_field', 'by_field'];
+
+    const resp = await callWithRequest('search', {
+      index: ML_RESULTS_INDEX_PATTERN,
+      size: 0,
+      body: {
+        query: {
+          bool: {
+            filter: [
+              ...criteriaFields.map(({ fieldName, fieldValue }) => {
+                return {
+                  term: {
+                    [fieldName]: fieldValue,
+                  },
+                };
+              }),
+              {
+                range: {
```

Comment:
I can only see this being used to get field values for record level results, so I think you should add an extra `result_type:record` term in here.

Reply:
Sorry, looking at https://github.com/elastic/kibana/pull/53839, should consider adding a term for `result_type:model_plot` or `result_type:record` depending on whether model plot has been enabled for the job.

Reply:
Check https://github.com/elastic/kibana/pull/53879/commits/72fb707e46a9ca16e9c4694d626dcf308d4b862d